waiter/src/waiter/core.clj

```diff
@@ -645,10 +645,14 @@
                                 (resolved-factory-fn (merge context component-config))))
                             custom-components)))
    :discovery (pc/fnk [[:curator curator]
-                       [:settings [:cluster-config name] [:zookeeper base-path discovery-relative-path] host port]
+                       [:settings
+                        [:cluster-config name]
+                        [:zookeeper base-path discovery-relative-path]
+                        host port {router-config {}}]
```

Comment:
Why is the syntax necessary for `router-config`?

Reply:
Because `router-config` is optional and if it's not present in settings, `pc/fnk` fails and Waiter won't start.